src/app/views/logs/Logs.jsx

```diff
@@ -1,124 +1,83 @@
             // TODO fix blank page when you go directly to a page number when there aren't enough pages for that to have any data
             // we should be redirecting them to just '/logs' instead
-            if (this.props.page && this.props.page !== "1") {
-                log.getAll((this.props.page - 1) * 10, this.state.pageSize, this.state.logsTimestamp).then(logRange => {
-                    this.setState({
-                        isFetchingLogs: false,
-                        logs: logRange,
-                    });
+            if (props.page && props.page !== "1") {
+                log.getAll((props.page - 1) * 10, pageSize, logsTimestamp).then(logRange => {
+                    setIsFetchingLogs(false);
+                    setLogs(logRange);
                 });
                 return;
             }
 
             if (count > 10) {
-                log.getAll(0, 10, this.state.logsTimestamp).then(logRange => {
-                    this.setState({
-                        isFetchingLogs: false,
-                        logs: logRange,
-                    });
+                log.getAll(0, 10, logsTimestamp).then(logRange => {
+                    setIsFetchingLogs(false);
+                    setLogs(logRange);
                 });
                 return;
             }
 
-            log.getAll(null, null, this.state.logsTimestamp).then(logs => {
-                this.setState({
-                    isFetchingLogs: false,
-                    logs,
-                });
+            log.getAll(null, null, logsTimestamp).then(logs => {
+                setIsFetchingLogs(false);
+                setLogs(null);
             });
         });
-    }
-
-    UNSAFE_componentWillReceiveProps(nextProps) {
-        if (this.props.page !== nextProps.page && nextProps.page === "1") {
-            this.setState({ isFetchingLogs: true });
-            log.getAll(0, 10, this.state.logsTimestamp).then(logRange => {
-                this.setState({
-                    isFetchingLogs: false,
-                    logs: logRange,
-                });
+    }, []);
+
+    useEffect(() => {
+        if (props.page === "1") {
+            setIsFetchingLogs(true);
+            log.getAll(0, 10, logsTimestamp).then(logRange => {
+                setIsFetchingLogs(false);
+                setLogs(logRange);
             });
             return;
-        }
-
-        if (this.props.page !== nextProps.page) {
-            this.setState({ isFetchingLogs: true });
-            log.getAll((nextProps.page - 1) * 10, this.state.pageSize, this.state.logsTimestamp).then(logRange => {
-                this.setState({
-                    isFetchingLogs: false,
-                    logs: logRange,
-                });
+        } else {
+            setIsFetchingLogs(true);
+            log.getAll((props.page - 1) * 10, pageSize, logsTimestamp).then(logRange => {
+                setIsFetchingLogs(false);
+                setLogs(logRange);
```

Comment:
@AnnaEmson so looking at lines 63-75 seems to be the problem. Not quite sure I did the refactoring here correctly.

From my understanding I changed the UNSAFE function to the useEffect hook and reference the props said function is dependent on. However the code then didn't like that nextProps was undefined so I reworked the conditionals to what I think it should be as useEffect doesn't make use of nextProps and useEffect should only run when props.page changes.

However after inspecting the props when going to the logs route the page prop is reading as undefined and throwing the entire thing off.

Basically I'm stuck and above is what I was able to figure out. Any ideas?

Reply:
so you confirmed the component `Logs.jsx` is used and you verified in production/ develop it is working?

Reply:
so when I investigated this now on prod
I got `Request URL: https://publishing.develop.onsdigital.co.uk/zebedee/collectionDetails/logs
Request Method: GET
Status Code: 404 `
so it looks like those APIs were removed or changed. This need to be checked. If it got broken needs to be fixed or is not needed anymore and we could clear all that stuff.  

So this is why  I think it would be good of Product Manager to verify if this functionality is used?

Reply:
So I went to http://localhost:8081/florence/logs and kinda just tried to fix the errors the console log spat out to me.

Reply:
I understand what you mean though that if this is broke on prod I guess refactoring this file is moot?